app/models/manageiq/providers/openstack/cloud_manager/cloud_volume_snapshot.rb

```diff
@@ -1,2 +1,136 @@
 class ManageIQ::Providers::Openstack::CloudManager::CloudVolumeSnapshot < ::CloudVolumeSnapshot
+  include SupportsFeatureMixin
+
+  supports :create_volume_snapshot
+  supports :update_volume_snapshot
+  supports :delete_volume_snapshot
+
+  def provider_object(connection)
+    connection.snapshots.get(ems_ref)
+  end
+
+  def with_provider_object
+    super(connection_options)
+  end
+
+  def self.create_volume_snapshot_queue(userid, cloud_volume, options = {})
+    ext_management_system = cloud_volume.try(:ext_management_system)
+    task_opts = {
+      :action => "creating volume snapshot in #{ext_management_system.inspect}\
+      for #{cloud_volume.inspect} with #{options.inspect}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => cloud_volume.class,
+      :instance_id => cloud_volume.id,
+      :method_name => 'create_volume_snapshot',
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => my_zone(ext_management_system),
+      :args        => [options]
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def self.create_volume_snapshot(cloud_volume, options = {})
+    raise ArgumentError, _("cloud_volume cannot be nil") if cloud_volume.nil?
+    ext_management_system = cloud_volume.try(:ext_management_system)
+    raise ArgumentError, _("ext_management_system cannot be nil") if ext_management_system.nil?
+
+    cloud_tenant = cloud_volume.cloud_tenant
+    snapshot = nil
+    options[:volume_id] = cloud_volume.ems_ref
+    ext_management_system.with_provider_connection(connection_options(cloud_tenant)) do |service|
+      snapshot = service.snapshots.new(options)
```

Comment:
I think fog supports also service.snapshots.create, doing the same thing as new&save ?